typegate/src/runtimes/prisma.ts

```diff
@@ -37,6 +42,59 @@ function isPrismaOperationMat(mat: Materializer): mat is PrismaOperationMat {
   return mat.name === "prisma_operation";
 }
 
+export interface PrismaRuntimeDS extends Omit<TGRuntime, "data"> {
+  data: PrismaRuntimeData;
+}
+
+registerHook("onPush", async (typegraph) => {
+  // TODO compare to previous version??
+  const runtimes = typegraph.runtimes.filter((rt) =>
+    rt.name === "prisma"
+  ) as unknown[] as PrismaRuntimeDS[];
+
+  if (runtimes.length === 0) {
+    return typegraph;
+  }
+
+  if (runtimes[0].data.migrations == null) {
+    if (runtimes.some((rt) => rt.data.migrations != null)) {
+      throw new Error(
+        "Migrations must be embedded for all the prisma runtimes",
+      );
+    }
+    return typegraph;
+  }
+
+  if (runtimes.some((rt) => rt.data.migrations == null)) {
+    throw new Error("Migrations must be embedded for all the prisma runtimes");
+  }
+
+  const tgName = typegraph.types[0].title;
+
+  for (const rt of runtimes) {
+    const { connection_string_secret, datamodel, migrations } = rt.data;
+
+    const datasource = makeDatasource(
+      envOrFail(tgName, connection_string_secret),
+    );
+
+    const res = nativeResult(
+      await native.prisma_deploy({
+        datasource,
+        datamodel,
+        migrations: migrations!,
+      }),
+    );
+
+    // TODO: log back to the user
```

Comment:
Indeed, we should aggregate all results from hook (e.g. allowing to warn when there was an upgrade in the format).